bluemira/geometry/parameterisations.py

```diff
@@ -986,6 +986,191 @@ def _make_control_points(p0, p3, theta0, theta3, l_start, l_end):
         return p1, p2
 
 
+class PictureFrameTools:
+    """
+    Curved Pictureframe parameterisation with straight (flat) inner limb
+
+    Parameters
+    ----------
+    var_dict: Optional[dict]
+        Dictionary with which to update the default values of the parameterisation.
+
+    """
+
+    @staticmethod
+    def _make_domed_leg(
+        axis,
+        x_out,
+        x_curve_start,
+        x_mid,
+        z_top,
+        z_mid,
+        flip=False,
+        r_c=0,
+    ):
+        """
+        Makes smooth dome for CP coils
+
+        Returns
+        -------
+        shape: BluemiraWire
+            CAD Wire of the geometry
+
+        """
+        # Define basic Top Curve (with no joint or corner transitions)
+
+        out_wire = []
+
+        # Define the basic main curve (with no joint or transitions curves)
+        alpha = np.arctan(0.5 * (x_out - x_curve_start) / abs(z_top - z_mid))
+        theta_leg_basic = 2 * (np.pi - 2 * alpha)
+        r_leg = 0.5 * (x_out - x_curve_start) / np.sin(theta_leg_basic / 2)
+        z_top_r_leg = z_top + r_leg if flip else z_top - r_leg
+        leg_centre = (x_out - 0.5 * (x_out - x_curve_start), 0, z_top_r_leg)
+
+        # Transitioning Curve
+        sin_a = np.sin(theta_leg_basic / 2)
+        cos_a = np.cos(theta_leg_basic / 2)
```

Comment:
Detail: multiplication is always faster than divisions (for all the `/ 2` before an after this line)

```suggestion
        cos_a = np.cos(0.5 * theta_leg_basic)
```